frontend/model/captureLogs.js

```diff
@@ -1,218 +1,87 @@
+// Furthermore, 'beforeunload' has implications for how the bfcache works.
+window.addEventListener('visibilitychange', event => sbp('appLogs/save').catch(e => {
+  console.error('Error saving logs during visibilitychange event handler', e)
+}))
 
-function getLogger (): Object {
-  if (!logger) {
-    logger = createLogger(config)
-    const previousEntries = JSON.parse(getItem('entries') ?? '[]')
-
-    // If `maxEntries` is changed in a release, this will discard oldest logs as necessary.
-    if (config.maxEntries < previousEntries.length) {
-      previousEntries.splice(0, previousEntries.length - config.maxEntries)
-    }
-    // Load the previous entries to sync the in-memory array with the local storage.
-    if (previousEntries.length) {
-      logger.entries.addAll(previousEntries)
-    }
-  }
-  return logger
-}
+// Enable logging to the server
+logServer(originalConsole)
 
-function setAppLogsFilter (filter: Array<string>) {
-  appLogsFilter = filter
-  // NOTE: Find a way to capture logs without messing up with log file location.
-  // console.log() doesnt include stack trace, so when logged, we can't access
-  // where the log came from (file name), which} difficults debugging if needed.
-  for (const level of loggingLevels) {
-    // $FlowFixMe
-    consoleCopy[level] = appLogsFilter.includes(level) ? logger[level] : noop
-  }
-}
-
-window.addEventListener('beforeunload', event => sbp('appLogs/save'))
-
-sbp('sbp/selectors/register', {
-  'appLogs/downloadOrShare': downloadOrShareLogs,
-  'appLogs/get' () { return getLogger()?.entries?.toArray() ?? [] },
-  'appLogs/save' () { getLogger()?.save() },
+export default (sbp('sbp/selectors/register', {
+  'appLogs/get' () { return logger?.entries.toArray() ?? [] },
+  async 'appLogs/save' () { await logger?.save() },
   'appLogs/pauseCapture': captureLogsPause,
   'appLogs/startCapture': captureLogsStart,
-  'appLogs/clearLogs': function (userID) {
+  async 'appLogs/clearLogs' (userID) {
     const savedID = identityContractID
     identityContractID = userID
-    try { clearLogs() } catch {}
+    try { await clearLogs() } catch {}
     identityContractID = savedID
-  },
-  // only log to server if we're in development mode and connected over the tunnel (which creates URLs that
-  // begin with 'https://gi' per Gruntfile.js)
-  'appLogs/logServer': process.env.NODE_ENV !== 'development' || !window.location.href.startsWith('https://gi')
-    ? noop
-    : function (level, stringifyMe) {
-      if (level === 'debug') return // comment out to send much more log info
-      const value = JSON.stringify(stringifyMe)
-      fetch(`${sbp('okTurtles.data/get', 'API_URL')}/log`, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify({ level, value })
-      }).catch(e => {
-        originalConsole.error(`[captureLogs] '${e.message}' attempting to log [${level}] to server:`, value)
-      })
```

Comment:
This code was really useful and now it's being removed. Could it please be restored?

I realize the `grunt dev --tunnel` flag might not work reliably anymore, but this useful code could still be adapted to work with other tunnels for capturing logs when testing on mobile.

Reply:
Sure, I can look into this. It was removed in part because it interfered with the SW logs somehow. IIRC, it wasn't this code but rather how it was called. I can try adding it back and see exactly what the issue was and how to address it.

Reply:
Hmm, I just noticed it wasn't removed, but seems to have been moved to `frontend/model/logServer.js`...